tests/test_network.py

```diff
@@ -124,6 +128,102 @@ def test_add_objective(factory, client):
     assert dataset.objective_key == objective.key
 
 
+@pytest.mark.parametrize('asset_name', [
+    'dataset',
+    #'objective',
+    #'algo',
+    #'aggregate_algo',
+    #'composite_algo',
+])
+@pytest.mark.parametrize('metadata,metadata_output,metadata_error', [
+    ({'foo': 'bar'}, {'foo': 'bar'}, {'foo': {"bar": "foo"}}),
+    (None, {}, {too_long: "bar"}),
+    ({}, {}, {"foo": too_long}),
+])
+def test_metadata(factory, client, asset_name, metadata, metadata_output, metadata_error):
```

Comment:
This test could be splitted in 2 tests: test_valid_metadata and test_invalid_metadata. The former one would use the asset_name / metadata / metadata_output as args, and the latter one would use just asset_name and metadata. 

In order to handle all types of bad metadata in the same way, you would have to set the bad metadata on the spec_dict directly.